classes/class-wc-connect-service-settings-store.php

```diff
@@ -85,32 +85,45 @@ public function update_account_settings( $settings ) {
 
 		public function get_selected_payment_method_id() {
 			$account_settings = $this->get_account_settings();
-			return intval( $account_settings[ 'selected_payment_method_id' ] );
+			return intval( $account_settings['selected_payment_method_id'] );
 		}
 
 		public function set_selected_payment_method_id( $new_payment_method_id ) {
 			$new_payment_method_id = intval( $new_payment_method_id );
 			$account_settings = $this->get_account_settings();
-			$old_payment_method_id = intval( $account_settings[ 'selected_payment_method_id' ] );
+			$old_payment_method_id = intval( $account_settings['selected_payment_method_id'] );
 			if ( $old_payment_method_id === $new_payment_method_id ) {
 				return;
 			}
-			$account_settings[ 'selected_payment_method_id' ] = $new_payment_method_id;
+			$account_settings['selected_payment_method_id'] = $new_payment_method_id;
 			$this->update_account_settings( $account_settings );
 		}
 
 		public function get_origin_address() {
 			$wc_address_fields = array();
-			$wc_address_fields[ 'company' ] = get_bloginfo( 'name' );
-			$wc_address_fields[ 'name' ] = wp_get_current_user()->display_name;
-			$base_location = wc_get_base_location();
-			$wc_address_fields[ 'country' ] = $base_location[ 'country' ];
-			$wc_address_fields[ 'state' ] = $base_location[ 'state' ];
-			$wc_address_fields[ 'address' ] = '';
-			$wc_address_fields[ 'address_2' ] = '';
-			$wc_address_fields[ 'city' ] = '';
-			$wc_address_fields[ 'postcode' ] = '';
-			$wc_address_fields[ 'phone' ] = '';
+			$wc_address_fields['company'] = get_bloginfo( 'name' );
+			$wc_address_fields['name'] = wp_get_current_user()->display_name;
+
+			$wc_countries = WC()->countries;
+			// WC 3.2 introduces ability to configure a full address in the settings
+			// Use it for address defaults if available
+			if ( method_exists( $wc_countries, 'get_base_address' ) ) {
+				$wc_address_fields['country'] = $wc_countries->get_base_country();
+				$wc_address_fields['state'] = $wc_countries->get_base_state();
+				$wc_address_fields['address'] = $wc_countries->get_base_address();
+				$wc_address_fields['address_2'] = $wc_countries->get_base_address_2();
+				$wc_address_fields['city'] = $wc_countries->get_base_city();
+				$wc_address_fields['postcode'] = $wc_countries->get_base_postcode();
+			} else {
+				$base_location = wc_get_base_location();
+				$wc_address_fields['country'] = $base_location['country'];
+				$wc_address_fields['state'] = $base_location['state'];
+				$wc_address_fields['address'] = '';
+				$wc_address_fields['address_2'] = '';
+				$wc_address_fields['city'] = '';
+				$wc_address_fields['postcode'] = '';
+			}
+			$wc_address_fields['phone'] = '';
```

Comment:
How about putting this line at the top before the if statement, with the rest of the array keys that are the same in both cases? Right below
`$wc_address_fields['name'] = wp_get_current_user()->display_name;`.